src/gui/editors/SongEditor.cpp

```diff
@@ -264,6 +285,48 @@ void SongEditor::loadSettings( const QDomElement& element )
 
 
 
+float SongEditor::getSnapSize() const
+{
+	// 1 Bar is the third value in the snapping dropdown
+	int val = -m_snappingModel->value() + 3;
+	// If proportional snap is on, we snap to finer values when zoomed in
+	if (m_proportionalSnap)
+	{
+		val = val - m_zoomingModel->value() + 3;
+	}
+	val = max(val, -6); // -6 gives 1/64th bar snapping. Lower values cause crashing.
+
+	if ( val >= 0 ){
+		return 1 << val;
+	}
+	else {
+		return 1.0 / ( 1 << -val );
+	}
+}
+
+QString SongEditor::getSnapSizeString() const
+{
+	int val = -m_snappingModel->value() + 3;
+	val = val - m_zoomingModel->value() + 3;
+	val = max(val, -6); // -6 gives 1/64th bar snapping. Lower values cause crashing.
+
+	if ( val >= 0 ){
+		int bars = 1 << val;
+		if ( bars == 1 ) { return QString("1 Bar");
+		else
+		{
+			QString( "%1 Bars" ).arg(bars);
```

Comment:
```suggestion
			return QString( "%1 Bars" ).arg(bars);
```
Scala really spoils me.